AK/StringView.h

```diff
@@ -36,16 +37,18 @@ class StringView {
 public:
     using ConstIterator = const char*;
 
-    StringView() {}
+    StringView() { }
     StringView(const char* characters, size_t length)
         : m_characters(characters)
         , m_length(length)
     {
+        ASSERT(!__builtin_add_overflow_p((uintptr_t)characters, length, (uintptr_t)0));
```

Comment:
Can we add a helper in `Checked.h` for this?
Something like `bool Checked<T>::addition_would_overflow(U, V)`

Reply:
Sure. But I've found that if I write it the the way `Checked::multiplication_would_overflow()` is currently written, GCC is unable to optimize out the intermediary variable (so it stores the flag in a register and then branches on that register, rather than branching on the flag directly). Would you be fine with using `__builtin_..._overflow_p` intrinsics in `Checked::..._would_overflow()` (instead of defining them in terms of other `Checked` operations)?

Reply:
Of course :)